testData/actions/generation/generator/UiComponentGridDataProviderGenerator/generateDataProviderWithInjectedGetListQuery/GridDataProvider.php

```diff
@@ -99,9 +99,14 @@ public function getData(): array
             return $this->loadedData;
         }
         $this->loadedData = parent::getData();
+        $itemsById = [];
 
         foreach ($this->loadedData['items'] as $item) {
-            $this->loadedData['items'][$item['entity_id']] = $item;
+            $itemsById[(int)$item['entity_id']] = $item;
+        }
+
+        if ($this->request->getParam('entity_id')) {
+            $this->loadedData['entity'] = $itemsById[(int)$this->request->getParam('entity_id')];
```

Comment:
In my opinion, isn't the best choice to parse/fetch all the items (this would be a performance killer), in order to get the needed one, when we're opening the Form. 
Please correct me if I'm wrong.

Reply:
@eduard13 You are right. But here is always one item fetched. Magento cares about it for us by adding filter by id field to the search criteria builder when we deal with the UI Component form.